sdk/textanalytics/ai-text-analytics/src/analyzeLro.ts

```diff
@@ -0,0 +1,248 @@
+    private pollOptions: {
+      includeStatistics?: boolean;
+    },
+    private documents: TextDocumentInput[],
+    private tasks: JobManifestTasks
+  ) {}
+  async sendInitialRequest(): Promise<LroResponse<PagedAnalyzeActionsResult>> {
+    const { span, updatedOptions: finalOptions } = createSpan("TextAnalyticsClient-beginAnalyze", {
+      ...this.baseOptions,
+      ...this.initOptions
+    });
+    try {
+      const { flatResponse, rawResponse } = await getRawResponse(
+        (paramOptions) =>
+          this.client.analyze({
+            body: {
+              analysisInput: { documents: this.documents },
+              tasks: this.tasks,
+              displayName: this.initOptions.displayName
+            },
+            ...paramOptions
+          }),
+        finalOptions
+      );
+      return {
+        flatResponse: flatResponse as PagedAnalyzeActionsResult,
+        rawResponse
+      };
+    } catch (e) {
+      const exception = handleInvalidDocumentBatch(e);
+      span.setStatus({
+        code: SpanStatusCode.ERROR,
+        message: exception.message
+      });
+      throw exception;
+    } finally {
+      span.end();
+    }
+  }
+  async sendPollRequest(path: string): Promise<LroResponse<PagedAnalyzeActionsResult>> {
+    return sendGetRequest(
+      this.client,
+      analyzeStatusOperationSpec,
+      "AnalyzeStatus",
+      { ...this.baseOptions, ...this.pollOptions },
+      path
+    ) as Promise<LroResponse<PagedAnalyzeActionsResult>>;
+  }
+}
+
+/**
+ * @internal
+ */
+export function isAnalyzeDone(response: unknown): boolean {
+  const castResponse = response as GeneratedClientAnalyzeStatusResponse;
+  switch (castResponse.status) {
+    case "notStarted":
+    case "running":
+      return false;
+    default: {
+      return true;
+    }
+  }
+}
+
+/**
+ * @internal
+ */
+export function processAnalyzeResult<TOptions extends OperationOptions>(
+  // eslint-disable-next-line @azure/azure-sdk/ts-use-interface-parameters
+  client: GeneratedClient,
+  documents: TextDocumentInput[],
+  options: TOptions
+): (result: unknown, state: AnalyzeActionsOperationState) => PagedAnalyzeActionsResult {
+  const pagedResult: PagedResult<
+    TOptions,
+    GeneratedClientAnalyzeStatusResponse,
+    AnalyzeActionsResult
+  > = {
+    sendGetRequest: (path: string, optionsParam: TOptions) =>
+      sendGetRequest(client, analyzeStatusOperationSpec, "AnalyzeStatus", optionsParam, path).then(
+        (response) => response.flatResponse as GeneratedClientAnalyzeStatusResponse
+      ),
+    buildPage: (flatResponse: GeneratedClientAnalyzeStatusResponse) => {
+      if (flatResponse) {
+        return createAnalyzeActionsResult(flatResponse, documents);
+      } else {
+        throw new Error("Analyze action has succeeded but there are no results!");
+      }
+    }
+  };
+  return (_result: unknown, state: AnalyzeActionsOperationState): PagedAnalyzeActionsResult => {
+    const pollingURL = (state as any).pollingURL;
```

Comment:
there is an internal one in core-lro:https://github.com/Azure/azure-sdk-for-js/blob/fbc0cf2ec27afea45571b324605e6e71cb1a41f3/sdk/core/core-lro/src/lroEngine/models.ts#L97 but it would be overkill to export it. Remember we agreed on making `pollingURL` an undocument property in the state type?

Reply:
Sure, I was thinking more of something inline like `state as any as { pollingURL: string }` but I suppose that doesn't buy us much